### ISSUE TITLE
How to create duplex UMIs with AnnotateBamWithUmis

### ISSUE THREAD

**Issue by birnbera** [fulcrumgenomics/fgbio#656]
I am working with data that uses two UMIs for paired end reads. One UMI was included as part of index 1 and the other as part of index 2. I'd like to annotate the RX field in my BAM file with both UMIs with a dash between, as in `NNNNNNNN-NNNNNNNN`. I see that `CorrectUmis` can handle duplex UMIs, such that it looks for the consensus sequence independently for each half. What would be the best way annotate a BAM with duplex UMIs in this case? It seems with `AnnotateBamWithUmis` you end up replacing the contents of the RX field when you run it twice.

**Comment by nh13** [comment]
@birnbera 

You're right that `AnnotateBamWithUmis` takes only one FASTQ.  Can you try the proposed update in PR#657 where I added support for _multiple_ FASTQs?

**Comment by birnbera** [comment]
Thanks for looking into this and coming up with a solution! I'm happy to run my dataset through and report back. It looks like the PR is still building in Travis CI. Do you know if there's anywhere that I can grab the built jar from the test build when it's done? I don't have a scala build environment set up. It would be preferable if there's a way to get a jar to test with.

**Comment by nh13** [comment]
@birnbera https://github.com/fulcrumgenomics/fgbio/releases/tag/nh_annotate_multiple_fastqs_tag.  Download the JAR there and I can then delete the release.

**Comment by birnbera** [comment]
Perfect. Just ran it and the result is as advertised. I used the resulting annotated bam (following sorting and addition of read groups) with `picard UmiAwareMarkDuplicatesWithMateCigar` and encountered no issues. Thanks again!

**Comment by nh13** [comment]
@birnbera you can also try using `GroupReadsWithUmi -> CallDuplexConsensusReads -> FilterConsensusReads` if you have duplex data versus just using the UMIs to mark duplicates.  I'll close this issue once the PR is merged.

**Comment by birnbera** [comment]
No issues detected after running through the suggested pipeline of additional tools.

**Comment by birnbera** [comment]
While I have your attention, I'd like to ask about the use case for `GroupReadsWithUmi` as opposed to a tool like `UmiAwareMarkDuplicatesWithMateCigar`. It seems that when you're using UMI's it's because you're really interested in the ratio between reads at different locations and you need to be careful with PCR duplicates. In the case of `GroupReadsWithUmi` it so aggressively filters out reads that I'd expect the result to be very biased depending on which reads pass QC. What use case do you envision and have you tested whether my hypothesis is correct for whatever it has been used for?

**Comment by nh13** [comment]
`UmiAwareMarkDuplicatesWithMateCigar` is mainly used to ensure that two or more reads from the same source molecule are not counted twice in downstream variant calling, but why not leverage the multiple observations for variant calling.  I am mostly interested in using UMIs to detect groups where reads that original from the same source molecule are in the same group so that I can then create a  _consensus_ from each group of reads.  The _consensus_ has much lower error rate and sources of bias (depending on the UMI scheme) so can enable ultra-accurate variant calling (SNVs/indels/STRs/etc.). 

> In the case of GroupReadsWithUmi it so aggressively filters out read.

Well you can turn off filters and some additional hard-coded filters are being relaxed (see #648) but we'd welcome an [opportunity to collaborate](https://www.fulcrumgenomics.com/contact/).

**Comment by birnbera** [comment]
Got it. My use case is different but it makes sense to do more aggressive quality filtering for variant detection. Thank you for your help on this!
